File: 02_스택, 큐, 덱/라이브 코딩/10866.cpp
// µ¦
#include <iostream>
#include <queue>

using namespace std;

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL); cout.tie(NULL);
	
	int n, x;
	deque<int> s;
	string input;

	cin >> n;
	while (n--) {
		cin >> input;
		if (input == "push_front") {
			cin >> x;
			s.push_front(x);
		}
		else if (input == "push_back") {
			cin >> x;
			s.push_back(x);
		}
		else if (input == "pop_front") {
			if (s.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << s.front() << '\n';
				s.pop_front();
			}
		}
		else if (input == "pop_back") {
			if (s.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << s.back() << '\n';
				s.pop_back();
			}
		}
		else if (input == "size") {
			cout << s.size() << '\n';
		}
		else if (input == "empty") {
			cout << s.empty() << '\n';
		}
		else if (input == "front") {
			if (s.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << s.front() << '\n';
			}
		}
		else if (input == "back") {
			if (s.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << s.back() << '\n';
			}
		}
	}
}

File: 02_스택, 큐, 덱/라이브 코딩/10866.py
from collections import deque
N = int(input())
deq = deque()
ans = ""

for _ in range(N):
    inp = input().split()
    command = inp[0]

    if command == 'push_front':
        deq.appendleft(int(inp[1]))
    elif command == 'push_back':
        deq.append(int(inp[1]))
    elif command == 'pop_front':
        if len(deq) == 0:
            ans += str(-1) + '\n'
        else:
            ans += str(deq.popleft()) + '\n'
    elif command == 'pop_back':
        if len(deq) == 0:
            ans += str(-1) + '\n'
        else:
            ans += str(deq.pop()) + '\n'
    elif command == 'size':
        ans += str(len(deq)) + '\n'
    elif command == 'empty':
        if len(deq) == 0:
            ans += str(1) + '\n'
        else:
            ans += str(0) + '\n'
    elif command == 'front':
        if len(deq) == 0:
            ans += str(-1) + '\n'
        else:
            ans += str(deq[0]) + '\n'
    else:
        if len(deq) == 0:
            ans += str(-1) + '\n'
        else:
            ans += str(deq[-1]) + '\n'

print(ans)